### PR TITLE
gh-119521: Rename IncompleteInputError to _IncompleteInputError and remove from stable ABI

### PR DESCRIPTION
**Pull request by pablogsal**
* Issue: gh-119521

----
📚 Documentation preview 📚: https://cpython-previews--119680.org.readthedocs.build/

### CODE REVIEW

**Review by JelleZijlstra — COMMENTED** [review]
Should we use `__IncompleteInputError__`? There are currently no other classes with a single underscore in the builtins, but there are some other internal names that use dunders (e.g., `__import__`, `__build_class__`).

**Comment by pablogsal** [comment]
That would be a bit weird IMHO because those dunder methods are hooks or functions, not classes

File: Include/pyerrors.h [hunk 1]
```diff
@@ -108,7 +108,9 @@ PyAPI_DATA(PyObject *) PyExc_NotImplementedError;
 PyAPI_DATA(PyObject *) PyExc_SyntaxError;
 PyAPI_DATA(PyObject *) PyExc_IndentationError;
 PyAPI_DATA(PyObject *) PyExc_TabError;
+#if !defined(Py_LIMITED_API)
 PyAPI_DATA(PyObject *) PyExc_IncompleteInputError;
```
Comment (zooba) [inline Include/pyerrors.h]: Does it need to be exported at all? An extension can always do `PyObject_GetAttrString(Py_GetBuiltins(), "_IncompleteInputError"))` (or whatever the "GetBuiltins" call actually is) if it needs a reference.

Removing from the limited API is definitely the right thing. Luckily we caught it pre-release.

Comment (pablogsal) [reply Include/pyerrors.h]: We are using it in the parser so it needs to be exported in the headers somehow. Or do you mean we can just use "extern" for it?

Comment (zooba) [reply Include/pyerrors.h]: Yeah, just `extern` would mean it's available in builds that are also building its source file, while `PyAPI_` means it's exported for use by other apps that link to our binaries. (I think the distinction is more strictly enforced on Windows than POSIX, but we can at least capture our intent.)

**Comment by encukou** [comment]
@pablogsal Do you plan to get back to this? Should I take over?

**Comment by pablogsal** [comment]
I think the only thing missing is https://github.com/python/cpython/pull/119680#discussion_r1622326912

I can try to do it this week, but feel free to take over if you have free cycles

**Comment by encukou** [comment]
How's this? It won't expose a public pointer at all, instead there's `#define PyExc_IncompleteInputError ((PyObject *)(&_PyExc_IncompleteInputError))` in an internal header?

**Comment by pablogsal** [comment]
LGTM

**Comment by encukou** [comment]
`test_peg_generator` needs this exported; `extern` won't cut it :(

**Pull request merged**